src/DependencyInjection/Compiler/HttpCachePass.php

```diff
@@ -22,42 +21,39 @@ class HttpCachePass implements CompilerPassInterface
 {
     public function process(ContainerBuilder $container)
     {
-        $this->processCacheManager($container);
-        $this->setProxyClient($container);
+        $this->processHttpDispatcher($container);
+        $this->processVarnishProxyClient($container);
     }
 
-    private function processCacheManager(ContainerBuilder $container)
+    private function processHttpDispatcher(ContainerBuilder $container)
     {
-        if (!$container->hasDefinition('ezplatform.http_cache.cache_manager')) {
+        if (!$container->hasDefinition('fos_http_cache.proxy_client.varnish.http_dispatcher')) {
             return;
         }
 
-        // Remove so 7.0 & 6.x kernels skip HttpCachePass
-        $container->removeAlias('ezpublish.http_cache.cache_manager');
+        $httpDispatcher = $container->findDefinition('fos_http_cache.proxy_client.varnish.http_dispatcher');
+        $httpDispatcher->replaceArgument(0, []);
```

Comment:
still needed? if so add comment on why it's here maybe, or make method name elf describing

Reply:
I have changed code to make it more explicit and also added explanation for it.